format/Message.fbs

```diff
@@ -92,6 +92,31 @@ union Type {
 }
 
 /// ----------------------------------------------------------------------
+/// The possible types of a vector
+
+enum VectorType: short {
+  /// used in List type, Dense Union and variable length primitive types (String, Binary)
+  OFFSET,
+  /// actual data, either wixed width primitive types in slots or variable width delimited by an OFFSET vector
+  DATA,
+  /// Bit vector indicating if each value is null
+  VALIDITY,
+  /// Type vector used in Union type
+  TYPE
+}
+
+/// ----------------------------------------------------------------------
+/// represents the physical layout of a buffer
+/// buffers have fixed width slots of a given type
+
+table VectorLayout {
+  /// the width of a slot in the buffer (typically 1, 8, 16, 32 or 64)
+  bit_width: short;
```

Comment:
it is more useful in cases where the bit_width is less definitive:
 - what is the dictionary id size?
 - what's the decimal byte size?